### PR TITLE
Issue #1983: Add a dxvk-nvapi verb

### PR DESCRIPTION
**Pull request by ThisNekoGuy**
Please let me know if I overlooked anything or otherwise did something wrong; I _**think**_ I did this right but I'm not sure.

dxvk-nvapi is (mostly) a replacement for the physx verb, because it supports physx calls, but I didn't want to touch the physx verb since I'd be unsure of the potential impact; though, it is pretty safe to say that the physx verb is otherwise deprecated (but do correct me if you know more than I do).

### CODE REVIEW

File: src/winetricks [hunk 1]
```diff
@@ -8215,6 +8215,69 @@ load_dxvk()
...
+
+    _W_supported_overrides="nvapi nvapi64"
+    _W_invalid_overrides="$(echo "${_W_dll_overrides}" | awk -vvalid_overrides_regex="$(echo "${_W_supported_overrides}" | sed 's/ /|/g')" '{ gsub(valid_overrides_regex,""); sub("[ ]*",""); print $0 }')"
+    if [ "${_W_invalid_overrides}" != "" ]; then
+        w_die "parameter (4) unsupported dll override: '${_W_invalid_overrides}' ; supported dll overrides: ${_W_supported_overrides}"
+    fi
+
+    _W_package_dir="${_W_package_archive%.tar.gz}"
+    _W_package_version="v${_W_package_dir#*-v}"
+    w_warn "Please refer to ${_W_repository#*/} version ${_W_package_version} release notes... See: https://github.com/${_W_repository}/releases/tag/${_W_package_version}"
```
Comment (austin987) [inline src/winetricks:8237]: Why the difference from dxvk? They end up with the same output, but dxvk_nvapi sets the "v" prefix in the variable assignment, while dxvk sets it in the warning itself?

File: src/winetricks [hunk 2]
```diff
@@ -8215,6 +8215,69 @@ load_dxvk()
...
+    if [ "${_W_package_archive##*.}" = "zip" ]; then
+        w_try_unzip "${W_TMP}" "${W_CACHE}/${W_PACKAGE}/${_W_package_archive}"
+    else
+        w_try_cd "${W_TMP}"
+        w_try tar -zxf "${W_CACHE}/${W_PACKAGE}/${_W_package_archive}"
+    fi
+
+    for _W_dll in ${_W_dll_overrides}; do
+        w_try mv "${W_TMP}/${_W_package_dir}/x32/${_W_dll}.dll" "${W_SYSTEM32_DLLS}/"
+    done
```
Comment (austin987) [inline src/winetricks:8249]: Please add a newline here.

**Comment by ThisNekoGuy** [comment]
@austin987 Adding your requested newline(s) seems to displease the shellcheck github action

**Comment by austin987** [comment]
> @austin987 Adding your requested newline(s) seems to displease the shellcheck github action

It should be a new line without trailing spaces (currently there are 4 empty spaces).

**Comment by austin987** [comment]
Hi @ThisNekoGuy,

Thanks for fixing the CI checks.

So, took a slightly closer look, now that it looks more stable. I'm unclear on some of the changes. I'll make more detailed comments inline.

File: src/winetricks [hunk 3]
```diff
@@ -8207,14 +8208,78 @@ load_dxvk()
 {
     # https://github.com/doitsujin/dxvk
     _W_dxvk_version="$(w_get_github_latest_release doitsujin dxvk)"
-    _W_dxvk_version="${_W_dxvk_version#v}"
-    w_linkcheck_ignore=1 w_download "https://github.com/doitsujin/dxvk/releases/download/v${_W_dxvk_version}/dxvk-${_W_dxvk_version}.tar.gz"
+    _W_dxvk_version="v${_W_dxvk_version#v}"
+    w_linkcheck_ignore=1 w_download "https://github.com/doitsujin/dxvk/releases/download/${_W_dxvk_version}/dxvk-${_W_dxvk_version}.tar.gz"
```
Comment (austin987) [inline src/winetricks:8212]: Hi again @ThisNekoGuy. I think the original change was fine. I didn't see the explanation from your commit message, just the diff itself. My apologies for misunderstanding.

Please adjust the commit message though, to something like:
"dxvk: fix link to release notes"

(doesn't have to be exactly that, feel free to tweak/explain, but commit message should clearly indicate dxvk, not just a line number).

File: src/winetricks (hunk 1 above)
Comment (austin987) [reply src/winetricks:8238]: To clarify, it's okay if things differ. I wanted to know why. Ideally, the code should be shared as much as possible, but differences happen.

That said, changes to dxvk itself should be avoided (unless it's a bug fix, or a change to _share_ code).

File: src/winetricks (hunk 3 above)
Comment (ThisNekoGuy) [reply src/winetricks:8212]: [Like this?](https://github.com/Winetricks/winetricks/pull/2033/commits/e5f977eeafa4fc8ec32d9ab684d407af88f2c8c8)

File: src/winetricks (hunk 1 above)
Comment (ThisNekoGuy) [reply src/winetricks:8238]: The only change I made to DXVK was how the package version variable collected the version tag; the value is only used for referencing the release notes here, so... what's the issue?

**Pull request merged**